features/old/accounts/invoices.feature

```diff
@@ -36,3 +36,38 @@ Feature: Show invoices from account's page (#16015909)
     And I follow "2011-01-00000001"
     # TODO: Properly check if order is opened
     Then I should see "Add"
+
+  Scenario: Display years of invoices when there are invoices
+    Given a provider is logged in on 1st January 2011
+    Given a buyer "zoidberg" signed up to provider "foo.3scale.localhost"
+    And the provider is charging its buyers in prepaid mode
+    And an invoice of buyer "zoidberg" for January, 2011
+    And I go to the invoices issued by me
+    Then I should see the list of years with invoices have the following years:
+      | 0  |
+      | 2011 |
+
+  Scenario: Display the current year when there are no invoices
+    Given a provider is logged in on 1st January 2011
+    And the provider is charging its buyers in prepaid mode
+    And I go to the invoices issued by me
+    Then I should see the list of years with invoices have the following years:
+      | 0  |
+      | current_year |
```

Comment:
No No this will be 2023 not 2011.

if user don't have any invoice then we show current year i.e 2023 so this is correct .

![Uploading image.png…]()

Reply:
The steps ideally should describe exactly what the user sees. So, having `current_year` is confusing. It should be an actual year. Here @jlledom suggested `2011` because this is what specified in the first step: Given a provider is logged in **on 1st January 2011** - in bold is the time helper, that sets the current date to the specified one.

Also, `0` should not be there either, because this is not what the user sees, and is not the actual year.

Reply:
so user is seeing 2023 as screenshot say and if user is logged in on 2011 but he don't have any invoice which is a correct case. 

And 0 is shown if you see the screenshot first blank is shown in UI see this screenshot

current year here used because of a reason, because in 2024 also the test will be valid if we have user with no invoice.

![image](https://github.com/3scale/porta/assets/84672731/088191c6-3a02-4539-868d-6ba26467017a)

Reply:
As @mayorova mentioned, the first step is freezing time at 2011, so `Time.now` will always be `1st January 2011`.

About the `0`, you can add it manually to `expected_years` in the step.